### PR TITLE
Specify patchable type

### PR DESCRIPTION
**Pull request by NobodysNightmare**
## OpenProject Work Packages
- https://community.openproject.org/work_packages/17420
- https://community.openproject.org/work_packages/17421
- https://community.openproject.org/work_packages/17422
## Changes
- type linked in WP rather than named
- new endpoints
  - type by id
  - all types
  - type by project
- project links to available types

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://feature-specify-patchable-type.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

**Comment by NobodysNightmare** [comment]
Should be ready-ish (for review)...

**Comment by hschink** [comment]
@NobodysNightmare I guess this PR needs a rebase.

**Comment by NobodysNightmare** [comment]
@hschink Indeed!

File: doc/apiv3-documentation.apib [hunk 1]
```diff
@@ -1645,6 +1659,220 @@ but are also limited to the projects that the current user is allowed to see.
...
+
++ Response 200 (application/hal+json)
+
+    [Types][]
+
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:** view work package (on any project)
```
Comment (ulferts) [inline doc/apiv3-documentation.apib]: There is the `manage_types` permission which should also allow you to do this.

File: doc/apiv3-documentation.apib [hunk 2]
```diff
@@ -1645,6 +1659,220 @@ but are also limited to the projects that the current user is allowed to see.
...
+                                {
+                                    "href": "/api/v3/types/2"
+                                }
+                            },
+                            "_type": "Type",
+                            "id": 2,
+                            "name": "Feature",
+                            "position": 2,
+                            "isDefault": false,
+                            "isMilestone": false
```
Comment (ulferts) [inline doc/apiv3-documentation.apib]: The example does not include all the attributes listed above:
- color
- createdAt
- updatedAt

File: doc/apiv3-documentation.apib [hunk 3]
```diff
@@ -1645,6 +1659,220 @@ but are also limited to the projects that the current user is allowed to see.
...
```
Comment (ulferts) [inline doc/apiv3-documentation.apib]: please see above

File: doc/apiv3-documentation.apib [hunk 4]
```diff
@@ -1645,6 +1659,220 @@ but are also limited to the projects that the current user is allowed to see.
...
+
++ Response 200 (application/hal+json)
+
+    [Type][]
+
++ Response 403 (application/hal+json)
+
+    Returned if the client does not have sufficient permissions.
+
+    **Required permission:** view work package (on any project)
```
Comment (ulferts) [inline doc/apiv3-documentation.apib]: see above (manage_types)

File: doc/apiv3-documentation.apib [hunk 5]
```diff
@@ -2025,7 +2258,6 @@ Note that due to sharing this might be more than the versions *defined* by that
                 },
                 "id": 1528,
                 "subject": "Develop API",
-                "type": "Feature",
```
Comment (ulferts) [inline doc/apiv3-documentation.apib:2028]: What about embedding this?

Comment (NobodysNightmare) [inline doc/apiv3-documentation.apib:2028]: You know that I create these examples by hand? ^^

Consequently I should just remove all embedding in the examples, however my strategy until now was to keep embedded content that was already there, but not introducing new embedded objects.

Comment (ulferts) [inline doc/apiv3-documentation.apib:2028]: Now you know why I am asking! ^^

I know that it is standard to let the server decide what to embed, but don't we want to provide some sort of hint regarding this?

Comment (NobodysNightmare) [inline doc/apiv3-documentation.apib:2028]: For the record: We discussed that I will remove embedding that is optional from the examples (more concise, less work for me :D )

But we also agreed that I will do that in a later PR, not in this one.

**Pull request merged**